### PR TITLE
[MNG-5001] Block the option to set Mojo parameters marked as @readonly

### PR DESCRIPTION
**Pull request by slachiewicz**

### CODE REVIEW

**Comment by rfscholte** [comment]
I would do all parameter validation at once, so you only have to loop one time. This means that readonly and required check are done by the same method.

**Comment by slachiewicz** [comment]
> I would do all parameter validation at once, so you only have to loop one time. This means that readonly and required check are done by the same method.

Could You suggest where this should be combined? That was my initial idea but during testing i found issues that not all test pass.
We have one validation in ValidatingConfigurationListener and also in DefaultMavenPluginValidator.validateParameters for maps.

**Comment by rfscholte** [comment]
I'd suggest to merge the code from https://github.com/apache/maven/blob/93e1f4a06b88cd29c8937a67b50791fcb1d07ff8/maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultMavenPluginManager.java#L766-L795 into https://github.com/apache/maven/blob/93e1f4a06b88cd29c8937a67b50791fcb1d07ff8/maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultMavenPluginManager.java#L712

**Comment by hboutemy** [comment]
core IT prepared in MNG-5001 branch https://github.com/apache/maven-integration-testing/tree/MNG-5001
@slachiewicz can you create a branch on Maven core to test and finish this feature, please?

**Comment by hboutemy** [comment]
@slachiewicz do you need help to finish this?

**Comment by slachiewicz** [comment]
Let me try again, thx for ITs.

File: maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultMavenPluginManager.java [hunk 1]
```diff
@@ -753,6 +755,37 @@ private void validateParameters( MojoDescriptor mojoDescriptor, PlexusConfigurat
...
+        {
+            PlexusConfiguration config = configuration.getChild( parameter.getName(), false );
+            if ( config != null && !parameter.isEditable() )
+            {
+                String value = config.getValue( null );
+                String defaultValue = config.getAttribute( "default-value", null );
+                if ( value != null && defaultValue != null && !Objects.equals( defaultValue, value ) )
+                {
+                    invalidParameters.add( parameter );
+                    logger.error( "Setting read-only parameter '" + parameter.getName() + "' to " + value );
```
Comment (eolivelli) [inline maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultMavenPluginManager.java]: Please report mojo/plugin/execution id f possible

**Comment by hboutemy** [comment]
happy to help: changing core is not easy
and doing it myself won't help getting more contribution

**Comment by jira-importer** [comment]
Resolve #7147
